core/src/main/java/com/yandex/yoctodb/v1/mutable/segment/V1StoredIndex.java

```diff
@@ -48,12 +53,10 @@ public IndexSegment addDocument(
 
         final UnsignedByteArray value = values.iterator().next();
 
-        this.valueDocId.putIfAbsent(value, new LinkedList<>());
-        if (this.valueDocId.get(value).size() != 0) {
+        this.valueDocId.computeIfAbsent(value, v -> new LinkedList<>()).add(documentId);
+        if (this.valueDocId.get(value).size() > 1) {
```

Comment:
It might be slightly faster to check `isEmpty()` of the value returned by previous `computeIfAbsent()` call.